ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -237,80 +251,52 @@ private void cleanupExpiredInflightRequests() {
-        break;
       case Blob_Authorization_Failure:
-        updateOperationState(replica, RouterErrorCode.BlobAuthorizationFailure);
-        break;
+        return RouterErrorCode.BlobAuthorizationFailure;
       case Blob_Expired:
-        updateOperationState(replica, RouterErrorCode.BlobExpired);
-        break;
+        return RouterErrorCode.BlobExpired;
       case Blob_Not_Found:
-        updateOperationState(replica, RouterErrorCode.BlobDoesNotExist);
-        break;
-      case Partition_Unknown:
-        updateOperationState(replica, RouterErrorCode.UnexpectedInternalError);
-        break;
+        return RouterErrorCode.BlobDoesNotExist;
       case Disk_Unavailable:
       case Replica_Unavailable:
-        updateOperationState(replica, RouterErrorCode.AmbryUnavailable);
-        break;
+        return RouterErrorCode.AmbryUnavailable;
       default:
-        updateOperationState(replica, RouterErrorCode.UnexpectedInternalError);
-        break;
-    }
-    if (serverErrorCode != ServerErrorCode.No_Error) {
-      logger.trace("Replica {} returned an error {} for a delete request with response correlationId : {} ",
-          replica.getDataNodeId(), serverErrorCode, correlationId);
+        return RouterErrorCode.UnexpectedInternalError;
     }
   }
 
   /**
-   * Updates the state of the {@code DeleteOperation}. This includes two parts: 1) resolves the
-   * {@link RouterErrorCode} depending on the precedence level of the new router error code from
-   * {@code replica} and the current {@code resolvedRouterErrorCode}. An error code with a smaller
-   * precedence level overrides an error code with a larger precedence level. 2) updates the
-   * {@code DeleteOperation} based on the {@link RouterErrorCode}, and the source {@link ReplicaId}
-   * for which the {@link RouterErrorCode} is generated.
-   * @param replica The replica for which the RouterErrorCode was generated.
-   * @param error {@link RouterErrorCode} that indicates the error for the replica.
+   * Perform the necessary actions when a request to a replica fails.
+   * @param replicaId the {@link ReplicaId} associated with the failed response.
+   * @param exception the {@link RouterException} associated with the failed response.
    */
-  private void updateOperationState(ReplicaId replica, RouterErrorCode error) {
-    if (resolvedRouterErrorCode == null) {
-      resolvedRouterErrorCode = error;
-    } else {
-      if (getPrecedenceLevel(error) < getPrecedenceLevel(resolvedRouterErrorCode)) {
-        resolvedRouterErrorCode = error;
-      }
-    }
-    operationTracker.onResponse(replica,
-        resolvedRouterErrorCode == RouterErrorCode.OperationTimedOut ? TrackedRequestFinalState.TIMED_OUT
-            : TrackedRequestFinalState.FAILURE);
-    if (error != RouterErrorCode.BlobDeleted && error != RouterErrorCode.BlobExpired) {
+  void onErrorResponse(ReplicaId replicaId, RouterException exception) {
+    logger.info("Get exception: " + exception);
```

Comment:
I think we don't need log here because in most places where `onErrorResponse` is called, there are TRACE logs.